mathesar/api/ui/permissions/schema_role.py

```diff
@@ -0,0 +1,54 @@
+from django.db.models import Q
+from rest_access_policy import AccessPolicy
+
+from mathesar.models.base import Database, Schema
+from mathesar.models.users import DatabaseRole, Role, SchemaRole
+
+
+class SchemaRoleAccessPolicy(AccessPolicy):
+    # Anyone can view schema role as long as they have atleast a view access to that schema or its database
+    # Create Access is restricted to superusers or managers of the schema or the database it belongs to.
+    statements = [
+        {
+            'action': ['list', 'retrieve', 'create'],
+            'principal': '*',
+            'effect': 'allow',
+        },
+        # Only superuser or schema/database manager can delete the role
+        {
+            'action': ['destroy', 'update', 'partial_update'],
```

Comment:
Why do we allow updates and partial updates here (also for DB roles)? I'm pretty sure I set it up so that we don't allow editing roles, only adding or deleting them.

Reply:
We haven't implemented updating or partially updating a role but IMO it shouldn't concern permissions. It might mislead someone to think they would need some higher permission level to update a role which is not the case here.